File: examples/python/client_example.py
"""
Example Python client for the scraping platform.

Usage:
    python client_example.py
"""

import requests
import json

class ScrapingClient:
    def __init__(self, api_key: str, base_url: str = "http://localhost:8000"):
        self.api_key = api_key
        self.base_url = base_url
        self.headers = {
            "Authorization": f"Bearer {api_key}",
            "Content-Type": "application/json"
        }
    
    def start_crawl(self, url: str, template: str) -> dict:
        """Start a new crawling job."""
        payload = {
            "url": url,
            "template": template
        }
        response = requests.post(
            f"{self.base_url}/api/v1/crawl",
            headers=self.headers,
            json=payload
        )
        return response.json()
    
    def get_job_status(self, job_id: str) -> dict:
        """Get the status of a crawling job."""
        response = requests.get(
            f"{self.base_url}/api/v1/jobs/{job_id}",
            headers=self.headers
        )
        return response.json()

if __name__ == "__main__":
    client = ScrapingClient("your-api-key-here")
    result = client.start_crawl("https://example.com", "company_profile_v1")
    print(json.dumps(result, indent=2))

File: examples/typescript/client_example.ts
import { ScrapingAPI } from '../../generated/api-client';

/**
 * TypeScript client example for the revolutionary scraping platform
 */
class ScrapingClient {
    private apiClient: ScrapingAPI;

    constructor(apiKey?: string, baseURL: string = '/api') {
        this.apiClient = new ScrapingAPI(baseURL, apiKey);
    }

    async startCrawl(url: string): Promise<any> {
        try {
            const result = await this.apiClient.crawlUrl({
                url,
                extraction_rules: [
                    {
                        name: 'title',
                        method: 'css',
                        selector: 'h1, .title',
                        content_type: 'text',
                        required: true
                    },
                    {
                        name: 'links', 
                        method: 'css',
                        selector: 'a[href]',
                        content_type: 'links',
                        multiple: true
                    }
                ]
            });
            return result;
        } catch (error) {
            console.error('Error starting crawl:', error);
            throw error;
        }
    }

    async getDashboardData(): Promise<any> {
        try {
            const result = await this.apiClient.getDashboardData();
            return result;
        } catch (error) {
            console.error('Error getting dashboard data:', error);
            throw error;
        }
    }

    async getProxyStats(): Promise<any> {
        try {
            const result = await this.apiClient.getProxyStats();
            return result;
        } catch (error) {
            console.error('Error getting proxy stats:', error);
            throw error;
        }
    }
}

// Usage example
async function demo() {
    const client = new ScrapingClient();
    
    try {
        // Create API instance for health check
        const api = new ScrapingAPI();
        
        // Health check
        const health = await api.healthCheck();
        console.log('System health:', health);
        
        // Get dashboard data
        const dashboard = await client.getDashboardData();
        console.log('Dashboard:', dashboard);
        
        // Start a crawl
        const crawlResult = await client.startCrawl('https://example.com');
        console.log('Crawl result:', crawlResult);
        
    } catch (error) {
        console.error('Demo failed:', error);
    }
}

export { ScrapingClient, demo };
